unittests/Feature/FeatureModelTransaction.cpp

```diff
@@ -224,4 +224,238 @@ TEST_F(FeatureModelTransactionModifyTest, addFeatureToModelThenAboard) {
   EXPECT_FALSE(FM->getFeature("ab")); // Change should not be visible
 }
 
+//===----------------------------------------------------------------------===//
+//                    FeatureModelMergeTransaction Tests
+//===----------------------------------------------------------------------===//
+
+class FeatureModelMergeTransactionTest : public ::testing::Test {
+protected:
+  void SetUp() override {
+    FeatureModelBuilder B;
+    B.makeFeature<BinaryFeature>("a", true);
+    FM = B.buildFeatureModel();
+    assert(FM);
+  }
+
+  std::unique_ptr<FeatureModel> FM;
+};
+
+TEST_F(FeatureModelMergeTransactionTest, Simple) {
+  FeatureModelBuilder B;
+  B.makeFeature<BinaryFeature>("b", true);
+  auto FM2 = B.buildFeatureModel();
+  assert(FM);
+
+  auto FMMerged = mergeFeatureModels(*FM, *FM2);
+  assert(FMMerged);
+
+  Feature *F1 = FMMerged->getFeature("a");
+  EXPECT_TRUE(F1);
+  Feature *F2 = FMMerged->getFeature("b");
+  EXPECT_TRUE(F2);
+  EXPECT_EQ(Feature::FeatureKind::FK_ROOT, F1->getParentFeature()->getKind());
+  EXPECT_EQ(Feature::FeatureKind::FK_ROOT, F2->getParentFeature()->getKind());
+}
+
+TEST_F(FeatureModelMergeTransactionTest, Idempotence) {
+  size_t FMSizeBefore = FM->size();
+
+  auto FMMerged = mergeFeatureModels(*FM, *FM);
+  assert(FMMerged);
+
+  EXPECT_EQ(FMSizeBefore, FMMerged->size());
+}
+
+TEST_F(FeatureModelMergeTransactionTest, DifferentLocations) {
+  FeatureSourceRange::FeatureSourceLocation FSL1(2, 4);
+  FeatureSourceRange::FeatureSourceLocation FSL2(2, 30);
+  FeatureSourceRange FSR1("path", FSL1, FSL2);
+  // TODO there must be a way to pass a copy directly
+  FeatureSourceRange FSRC1(FSR1);
+  FM->getFeature("a")->addLocation(FSRC1);
```

Comment:
@vulder is there a way to pass the FeatureSourceRange here directly without saving it in a local variable before?

Reply:
Oh, boy. Not with the current impl. Just change the function `addLocation(FeatureSourceRange FSR)` without the `&`. The std::move() afterwards also makes not much sense in the current impl with & :laughing:

Then you can just use it like this:
```cpp
    FM->getFeature("a")->addLocation(FeatureSourceRange{"path", {2, 4}, {2, 30}});
```